contracts/RelayerIncentivize.template

```diff
@@ -75,31 +78,50 @@ contract RelayerIncentivize is IRelayerIncentivize, System, IParamSubscriber {
     }
     headerRelayersSubmitCount[headerRelayerAddr]++;
 
-    if (transferRelayersSubmitCount[packageRelayer]==0) {
-      transferRelayerAddressRecord.push(packageRelayer);
+    if (packageRelayersSubmitCount[packageRelayer]==0) {
+      packageRelayerAddressRecord.push(packageRelayer);
     }
-    transferRelayersSubmitCount[packageRelayer]++;
+    packageRelayersSubmitCount[packageRelayer]++;
 
     if (countInRound==ROUND_SIZE) {
-      emit LogDistributeCollectedReward(roundSequence, collectedRewardForHeaderRelayer, collectedRewardForTransferRelayer);
+      emit distributeCollectedReward(roundSequence, collectedRewardForHeaderRelayer, collectedRewardForTransferRelayer);
+
+      uint256 callerHeaderReward = distributeHeaderRelayerReward();
+      uint256 callerPackageReward = distributePackageRelayerReward();
 
-      distributeHeaderRelayerReward(packageRelayer);
-      distributeTransferRelayerReward(packageRelayer);
+      uint256 totalRewardForDistributionTrigger = callerHeaderReward.add(callerPackageReward);
+      if (!packageRelayer.send(totalRewardForDistributionTrigger)){
+        address payable systemPayable = address(uint160(SYSTEM_REWARD_ADDR));
+        systemPayable.transfer(totalRewardForDistributionTrigger);
+      }
+      emit rewardToDistributionTrigger(packageRelayer, totalRewardForDistributionTrigger);
 
```

Comment:
We still need to send rest balance to system reward contracts, don't we?

Reply:
No rest balance